crates/engine/tree/src/tree/mod.rs

```diff
@@ -2167,8 +2170,29 @@ where
         let hashed_state = HashedPostState::from_bundle_state(&output.state.state);
 
         let root_time = Instant::now();
-        let (state_root, trie_output) =
-            state_provider.state_root_with_updates(hashed_state.clone())?;
+        let mut state_root_result = None;
+        let persistence_in_progress = self.persistence_state.in_progress();
+        if !persistence_in_progress {
```

Comment:
let's inline this  and get rid of the tmp var

Reply:
temp var cuz it's used later for logging